modules/services/screen-locker.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.screen-locker;
+
+in {
+
+  options.services.screen-locker = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable screen locker for X session";
+    };
+
+    lockCmd = mkOption {
+      type = types.str;
+      description = "Locker command to run.";
+      example = literalExample "${pkgs.i3lock}/bin/i3lock -n -c 000000";
```

Comment:
Hmm, will this not expand the `${pkgs.i3lock}` part? I think escaping is necessary, i.e., `\${pkgs.i3lock}`.

Reply:
And the `literalExample` is not necessary :-)

Reply:
It didn't expand in the manual for some reason, but I'll change it.

Reply:
Hmm, maybe `literalExample` does something magic? :-)